bulbs/super_features/models.py

```diff
@@ -0,0 +1,81 @@
+from django.db import models
+
+import jsonfield
+from elasticsearch_dsl import field
+
+from bulbs.content.models import Content
+from bulbs.super_features.utils import get_superfeature_choices, get_data_serializer
+
+
+GUIDE_TO_HOMEPAGE = 'GUIDE_TO_HOMEPAGE'
+GUIDE_TO_ENTRY = 'GUIDE_TO_ENTRY'
+BASE_CHOICES = (
+    (GUIDE_TO_HOMEPAGE, 'Guide To Homepage'),
+    (GUIDE_TO_ENTRY, 'Guide To Entry'),
+)
+
+SF_CHOICES = get_superfeature_choices()
+
+
+class AbstractSuperFeature(models.Model):
+    notes = models.TextField(null=True, blank=True, default='')
+    internal_name = models.CharField(null=True, blank=True, max_length=255)
+    superfeature_type = models.CharField(choices=SF_CHOICES, max_length=255)
+    default_child_type = models.CharField(choices=SF_CHOICES, max_length=255, null=True, blank=True)
+    data = jsonfield.JSONField()
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def get_data_serializer(cls, name):
+        return get_data_serializer(name)
+
+    def validate_data_field(self):
+        Serializer = self.get_data_serializer(self.superfeature_type)  # NOQA
+        Serializer(data=self.data).is_valid(raise_exception=True)
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        return super(AbstractSuperFeature, self).save(*args, **kwargs)
+
+    def clean(self, *args, **kwargs):
+        self.validate_data_field()
+        return super(AbstractSuperFeature, self).clean(*args, **kwargs)
+
+
+class BaseSuperFeature(Content, AbstractSuperFeature):
+    parent = models.ForeignKey('self', blank=True, null=True)
+    ordering = models.IntegerField(blank=True, null=True, default=None)
+
+    class Meta:
+        unique_together = ('parent', 'ordering')
+
+    class Mapping(Content.Mapping):
+        parent = field.Integer()
```

Comment:
@mparent61 @benghaziboy DJES would infinitely recurse if i didn't add this. not sure if this is terrible or not though.